ADD: add FTDI JTAG debugger that works with FPGA Pmod headers

T-K-233 opened the pull request.

sequencer commented:
We may only accept KiCAD designs? cc @vowstar

vowstar reviewed (changes requested):
Thank you very much for your commit, but we noticed that you commuted the gerber file directly, but we hope that you can commit the original project file directly, and facilitate github CI to perform automatic DRC and ERC, which is very conducive to review.

We ask for:

- Since this is an open source project, please commit the original files of the project directly, and please use open source EDA tools, such as ``kicad``
- The project directory has CMakeList, please refer to https://github.com/chipsalliance/rocket-pcb/blob/main/pmod_microsd/CMakeLists.txt for easy implementation
  - Automatic ERC ``cmake --build ./build --target erc``
  - Automatic DRC ``cmake --build ./build --target drc``
  - Automatically generate production materials (especially when tagging) `` cmake --build ./build --target production``
This ensures that all operations are reproducible on github CI.
 - If it is a ``pmod`` module, please use the ``pmod_`` prefix to facilitate the script to quickly distinguish the board type and prepare for later project automation.

T-K-233 commented:
Thanks for the feedback!

1. The EDA tool I'm using to produce the project is LCEDA, the zh-CN version of [EasyEDA](https://easyeda.com/), which is also a free software. I've included the project files in the directory. Unfortunately, LCEDA does not support exporting to KiCAD, but it does support exporting to Altium, so I also included the exported Altium project files there for people's convenience. I'm not sure if there's way to then export to KiCAD from Altium. But I think these two tools already covers a) free software and b) widely-used industry software.

2. Due to reason in 1., I've added a CMake to indicate this project requres LCEDA instead. I'm not sure how to tell the automated build system to ignore this project though, and I would appreciate if there's some help on this.

3. Updated.

sequencer commented:
> The EDA tool I'm using to produce the project is LCEDA, the zh-CN version of [EasyEDA](https://easyeda.com/), which is also a free software. I've included the project files in the directory. Unfortunately, LCEDA does not support exporting to KiCAD, but it does support exporting to Altium, so I also included the exported Altium project files there for people's convenience. I'm not sure if there's way to then export to KiCAD from Altium. But I think these two tools already covers a) free software and b) widely-used industry software.


I may require follow @vowstar's suggestion to use KiCAD to contribute to this project:( 
As you can see there is an ongoing project `candelabra` for Xilinx FPGA. It is also designed with KiCAD as well.

After I get some spare time, I'd like to tag a specific version of KiCAD to rocket-pcb for any other using this project can easily access the schematic w/o requiring specific vendor tools, this is sued to reduce the maintain burden and keeping everything up-to-date ;)

The goal are:
- an aligned KiCAD design flow;
- DRC/LVS on CI;
- A Release flow for all boards;
- a share [library](https://github.com/chipsalliance/rocket-pcblib) for any boards in this project.

T-K-233 commented:
I see. Thanks. 

I'll convert the project to KiCAD when I got time then.
